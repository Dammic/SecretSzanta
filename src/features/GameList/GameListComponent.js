'use strict'
import React from 'react'

const GameListComponent = ({
    title,
    userNames
}) => {
    console.info(userNames)
    return (
        <div className="main-page">
            Hello {title}!
            <ul data-columns="2">
                {userNames.map((name) => {
                    return (
                        <li key={name} className="name">
                            {name}
                        </li>
                    )
                })}
            </ul>
        </div>
    )
}

export default GameListComponent