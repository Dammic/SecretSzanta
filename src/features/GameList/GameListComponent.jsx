import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import GameListItemComponent from './GameListItemComponent'
import TopNavbar from './TopNavbar/TopNavbar'
import PlayersList from './PlayersList/PlayersList'
import Chat from '../UIBox/Chat/Chat'

const GameListComponent = ({
    userName,
    rooms,
    onClick,
}) => {
    return (
        <div className="game-room">
            <div className="game-list-container">
                <div className="first-column">
                    <div className="game-list">
                        {map(rooms, room => (
                            <GameListItemComponent
                                key={room.roomId}
                                roomId={room.roomId}
                                roomName={room.roomName}
                                playerCount={room.playerCount}
                                onClick={onClick}
                            />
                        ))}
                    </div>
                </div>
                <div className="second-column">
                    <div className="utils-row">
                        <span className="ellipsis">Hello <b>{userName}</b>!</span>
                        <TopNavbar />
                    </div>
                    <PlayersList />
                    <Chat />
                </div>
            </div>
        </div>
    )
}

GameListComponent.propTypes = {
    userName: PropTypes.string,
    rooms: PropTypes.objectOf(PropTypes.objectOf(PropTypes.any)),
    onClick: PropTypes.func,
}
export default GameListComponent
