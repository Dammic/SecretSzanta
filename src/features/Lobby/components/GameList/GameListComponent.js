import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import { Icon } from '../../../Shared/Icon'
import { ControlButton } from '../../../Shared/Buttons'

import styles from './GameList.css'

export const GameListComponent = ({ rooms, onClick }) => (
    <div className={styles.gameList}>
        {map(rooms, ({ roomName, playersCount, roomId, maxPlayers }) => (
            <div key={roomName} className={styles.gameListElement}>
                <div className={styles.gameElementContainer}>
                    <div className={styles.roomName}>{roomName}</div>
                    <div className={styles.rightSide}>
                        <ControlButton onClick={() => onClick(roomId)} className={styles.button}>
                            join
                        </ControlButton>
                        <span className={styles.playerCount}>
                            {playersCount}/{maxPlayers}
                            <Icon name="fa-user" className={styles.icon} />
                        </span>
                    </div>
                </div>
            </div>
        ))}
    </div>
)

GameListComponent.propTypes = {
    rooms: PropTypes.objectOf(PropTypes.shape({
        roomName: PropTypes.string,
        playersCount: PropTypes.number,
        roomId: PropTypes.string,
        maxPlayers: PropTypes.number,
    })),
    onClick: PropTypes.func,
}
