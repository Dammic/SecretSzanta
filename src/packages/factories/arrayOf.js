import faker from 'faker'
import { times } from 'lodash'

export const arrayOf = (callback, amount, options = { seed: null }) => {
    if (options.seed) {
        faker.seed(options.seed)
    }

    const result = []
    times(amount, () => result.push(callback()))
    return result
}
