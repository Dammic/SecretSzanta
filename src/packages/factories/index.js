export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
