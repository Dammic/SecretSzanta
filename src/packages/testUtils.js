import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
