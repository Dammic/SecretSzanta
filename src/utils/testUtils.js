import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
