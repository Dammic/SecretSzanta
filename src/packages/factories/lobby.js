import faker from 'faker'
import { keyBy } from 'lodash'
import { arrayOf } from './arrayOf'

export const aRoom = (propsOverrides, options = { seed: null }) => {
    if (options.seed) {
        faker.seed(options.seed)
    }

    return {
        roomName: faker.random.words(),
        playerCount: faker.random.number({ min: 0, max: 10 }),
        roomId: faker.random.word(),
        ...propsOverrides,
    }
}

export const aRoomsList = (propsOverrides, options = { seed: null }) => {
    if (options.seed) {
        faker.seed(options.seed)
    }

    return keyBy(arrayOf(aRoom, 3, options), 'roomId')
}

