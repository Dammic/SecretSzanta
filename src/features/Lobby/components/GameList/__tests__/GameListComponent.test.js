import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { aRoomsList } from 'packages/factories'
import { GameListComponent } from '../GameListComponent'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        rooms: aRoomsList(null, { seed: 1 }),
        onClick: jest.fn(),
        ...propsOverrides,
    }
    const component = renderMethod(<GameListComponent {...props} />)
    return { props, component }
}

describe('<GameListComponent />', () => {
    it('matches snapshot - renders list of rooms', () => {
        const { props } = setupProps()
        expectShallowMatchingSnapshot(<GameListComponent {...props} />)
    })

    it('matches snapshot - doesnt render list of rooms if there are not any', () => {
        const { props } = setupProps({ rooms: {} })
        expectShallowMatchingSnapshot(<GameListComponent {...props} />)
    })
})
