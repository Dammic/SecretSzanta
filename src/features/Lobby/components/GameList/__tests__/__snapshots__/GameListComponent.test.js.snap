// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameListComponent /> matches snapshot - doesnt render list of rooms if there are not any 1`] = `
<div
  className="gameList"
/>
`;

exports[`<GameListComponent /> matches snapshot - renders list of rooms 1`] = `
<div
  className="gameList"
>
  <div
    className="gameListElement"
  >
    <div
      className="gameElementContainer"
    >
      <div>
        <div>
          Producer Solutions
           |
          1
          /9999
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          JOIN
        </button>
      </div>
    </div>
  </div>
  <div
    className="gameListElement"
  >
    <div
      className="gameElementContainer"
    >
      <div>
        <div>
          Compatible
           |
          4
          /9999
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          JOIN
        </button>
      </div>
    </div>
  </div>
  <div
    className="gameListElement"
  >
    <div
      className="gameElementContainer"
    >
      <div>
        <div>
          Yen Operations
           |
          9
          /9999
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          JOIN
        </button>
      </div>
    </div>
  </div>
</div>
`;
